### PR TITLE
fix(block): display correct header spacing when heading or description are present

### PR DESCRIPTION
**Pull request by aPreciado88**
**Related Issue:** [#9920](https://github.com/Esri/calcite-design-system/issues/9920)

## Summary
Update `block` component to display correct header spacing when `heading` or `description` properties are present.

### CODE REVIEW

File: packages/calcite-components/src/components/block/block.scss
Comment (driskull) [inline packages/calcite-components/src/components/block/block.scss:201]: [description] would also be an issue here because it isn't reflected either. Should this css be removed?

File: packages/calcite-components/src/components/block/block.tsx
Comment (driskull) [inline packages/calcite-components/src/components/block/block.tsx:419]: should this be `!!(heading || description)`?

```suggestion
      <header class={{ [CSS.header]: true, [CSS.headerPadding]: !!(heading || description) }} id={IDS.header}>
```

**Review by driskull — CHANGES_REQUESTED** [review]
Changes look good, just missing a few things 👍

File: packages/calcite-components/src/components/block/block.scss
Comment (driskull) [inline packages/calcite-components/src/components/block/block.scss:26]: nitpick: maybe this should be named something like `.header--present` or `.header--enabled`?

**Comment by aPreciado88** [comment]
@driskull I noticed that if I update the `heading` prop in `block.tsx` to `@Prop({ reflect: true }) heading!: string;` then the css will work, without any of the changes above. Is there a reason why it's declared as `@Prop() heading!: string;` ? 
`description` is also declared as `@Prop() description: string;`. Should the fix be to add `{ reflect: true }` to both props?

**Comment by driskull** [comment]
@aPreciado88 no, we don't want to reflect all properties. In cases where the property is just content , we don't want to reflect. https://github.com/Esri/calcite-design-system/tree/dev/packages/calcite-components/conventions#reflecting-to-attributes

**Comment by aPreciado88** [comment]
@driskull I updated with the requested changes.

**Review by driskull — APPROVED** [review]
@aPreciado88 can we add a e2e test that checks for the css class when a heading or description is present via their properties?

Changes look good to me. Lets wait for @jcfranco to review before merging.

File: packages/calcite-components/src/components/block/block.scss
Comment (jcfranco) [inline packages/calcite-components/src/components/block/block.scss:26]: Nitpick: I believe we use `.<part>--has-text` in other components for similar contexts.

**Review by jcfranco — APPROVED** [review]
Once we have screenshot test coverage, this is good to merge! 🚀

File: packages/calcite-components/src/components/block/block.e2e.ts
Comment (jcfranco) [inline packages/calcite-components/src/components/block/block.e2e.ts:327]: This is fine for now until we add extra checks to ensure styles use reflected props. After that, we should rely on screenshot tests. From the assertion perspective, it is not guaranteed that changes in the CSS class will be detected by this test, meaning its implementation could change without causing a test failure. cc @driskull

Comment (aPreciado88) [reply packages/calcite-components/src/components/block/block.e2e.ts:327]: @jcfranco Should I create a follow up issue to add a screenshot test?

Comment (aPreciado88) [reply packages/calcite-components/src/components/block/block.e2e.ts:327]: @jcfranco I created this follow up [issue](https://github.com/Esri/calcite-design-system/issues/10084) to remove this E2E test once the reflected prop checks are in place.

**Pull request merged**
